### PR TITLE
Log out features

### PR DESCRIPTION
**Pull request by abdelrahmenAyman**
Log out User model,Log out controller, edit profile User Model , are all tested successfully.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -17,16 +15,18 @@
         <activity
```
Comment (KEMDAK) [inline app/src/main/AndroidManifest.xml:15]: make the Login activity the luncher activity

File: app/src/main/java/org/ieeeguc/ieeeguc/controllers/MainActivity.java [hunk 2]
```diff
@@ -1,21 +1,110 @@
...
         setContentView(R.layout.activity_main);
 
-        /* loading the logged in user and the token from the shared preferences */
+        // Sets the class to be a listener.
+        navigationView = (NavigationView) findViewById(R.id.navigation_view);
+        navigationView.setNavigationItemSelectedListener(this);
+
     }
+
+    private void logout() {
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/controllers/MainActivity.java]: add the method documentation

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 3]
```diff
@@ -1,83 +1,230 @@
...
+    private String email;
+    private Gender gender;
+    private Date birthdate;
+    private String ieeeMembershipID;
+    private int committeeID;
+    private String committeeName;
+    private String phoneNumber;
+    private JSONObject settings;
+
+    public User(){}
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: remove the empty constructor

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 4]
```diff
@@ -1,83 +1,230 @@
...
+            Request request = new Request.Builder().put(RequestBody.create(MediaType.parse("application/json"),
+                    new JSONObject(body).toString()))
+                    .addHeader("Authorization", token)
+                    .addHeader("user_agent", "Android")
+                    .url("http://ieeeguc.org/api/user").build();
+
+            client.newCall(request).enqueue(new Callback() {
+                @Override
+                public void onFailure(Call call, IOException e) {
+
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java:131]: add `call.close();`

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 5]
```diff
@@ -1,83 +1,230 @@
...
+                        try {
+                            JSONObject responseBody = new JSONObject(response.body().toString());
+                            httpResponse.onFailure(statusCode, responseBody);
+                        } catch (JSONException e) {
+
+                            httpResponse.onFailure(500, null);
+                        }
+
+                    }
+
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java:172]: add `response.close();`

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 6]
```diff
@@ -1,83 +1,230 @@
...
+                    .build();
+
+
+
+
+            ok.newCall(request).enqueue(new Callback() {
+                @Override
+
+                public void onFailure(Call call, IOException e) {
+                    call.cancel();
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java]: add `httpResponse.onFailure(-1, null);`

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java [hunk 7]
```diff
@@ -1,83 +1,230 @@
...
+                        {
+                            httpResponse.onSuccess(code,j);
+                        }
+                        else
+                        {
+                            httpResponse.onFailure(code,j);
+                        }
+                    } catch (JSONException e) {
+                        httpResponse.onFailure(500,null);
+                    }
```
Comment (KEMDAK) [inline app/src/main/java/org/ieeeguc/ieeeguc/models/User.java:222]: add `response.close();`

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -10,7 +10,14 @@
     <string name="error_invalid_password">This password is too short</string>
     <string name="error_incorrect_password">This password is incorrect</string>
     <string name="error_field_required">This field is required</string>
+    <string name="failed_logout_internet_problem">Failed to logout, please check your internet Connection!</string>
+    <string name="failed_logout_server_error">Sorry something went wrong with the server,please try again later!</string>
```
Comment (KEMDAK) [inline app/src/main/res/values/strings.xml]: change its name to "error_server_down"

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -10,7 +10,14 @@
     <string name="error_invalid_password">This password is too short</string>
     <string name="error_incorrect_password">This password is incorrect</string>
```
Comment (KEMDAK) [inline app/src/main/res/values/strings.xml]: remove the incorrect password, and add a new string that indicates incorrect credentials

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -10,7 +10,14 @@
     <string name="error_invalid_password">This password is too short</string>
     <string name="error_incorrect_password">This password is incorrect</string>
     <string name="error_field_required">This field is required</string>
+    <string name="failed_logout_internet_problem">Failed to logout, please check your internet Connection!</string>
```
Comment (KEMDAK) [inline app/src/main/res/values/strings.xml]: change its name to "error_connection"

**Review by KEMDAK — CHANGES_REQUESTED** [review]
Thanks @darkCrakin and @rehamayman96 both of you did a great job on your first sprint i hope you learned something new. Keep it up 👍 

@rehamayman96 your code is clean and well structured. good job.

@darkCrakin try to be systematic in your spacing :D

File: app/src/main/java/org/ieeeguc/ieeeguc/models/User.java (hunk 4 above)
Comment (abdelrahmenAyman) [reply app/src/main/java/org/ieeeguc/ieeeguc/models/User.java:131]: wait you mean call.cancel()? there is no method can be called on the Call object name close().

Comment (abdelrahmenAyman) [reply app/src/main/java/org/ieeeguc/ieeeguc/models/User.java:131]: I added call.cancel() until you clarify.

Comment (KEMDAK) [reply app/src/main/java/org/ieeeguc/ieeeguc/models/User.java:131]: Tammam

Comment (abdelrahmenAyman) [reply app/src/main/java/org/ieeeguc/ieeeguc/models/User.java:131]: el mafrood el branch keda gahez wi kol el changes were applied

**Review by KEMDAK — APPROVED** [review]

**Pull request merged**